examples/draw_turtle.rs

```diff
@@ -0,0 +1,323 @@
+    draw_leg(&mut turtle);
+
+    turtle.right(54.5);
+    turtle.forward(SIZE * 8.0);
+
+    draw_neck(&mut turtle);
+
+    turtle.right(172.6);
+    turtle.forward(SIZE * 40.0);
+    turtle.right(110.0);
+
+    draw_head(&mut turtle);
+
+    turtle.left(128.0);
+    turtle.forward(SIZE * 15.0);
+
+    draw_eye(&mut turtle);
+
+    turtle.set_fill_color(BODY_COLOR);
+
+    turtle.left(175.0);
+    turtle.forward(SIZE * 43.0);
+
+    draw_right_highlight(&mut turtle);
+
+    turtle.right(18.0);
+    turtle.forward(SIZE * 37.0);
+    turtle.set_heading(180.0);
+
+    draw_middle_highlight(&mut turtle);
+
+    turtle.left(24.0);
+    turtle.forward(SIZE * 36.0);
+    turtle.set_heading(180.0);
+
+    draw_left_highlight(&mut turtle);/**/
+}
+
+fn draw_shell(turtle: &mut Turtle) {
+    turtle.set_fill_color(SHELL_COLOR);
+    turtle.begin_fill();
+
+    for _ in 0..180 {
+        turtle.forward(SIZE);
+        turtle.right(1.0);
+    }
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 3.0);
+        turtle.right(1.0);
+    }
+
+    turtle.set_speed(5);
+    let d = SIZE * 360.0 / std::f64::consts::PI;
+    turtle.forward(d - d / 3.0);
+    turtle.set_speed(10);
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 3.0);
+        turtle.right(1.0);
+    }
+
+    turtle.end_fill();
+}
+
+fn draw_tail(turtle: &mut Turtle) {
+    turtle.set_fill_color(BODY_COLOR);
+    turtle.begin_fill();
+
+    turtle.left(90.0);
+    turtle.forward(SIZE);
+
+    for _ in 0..45 {
+        turtle.forward(SIZE / 3.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 3.0);
+
+    for _ in 0..25 {
+        turtle.forward(SIZE / 3.0);
+        turtle.left(6.0);
+    }
+
+    turtle.forward(SIZE * 9.0);
+    turtle.right(6.0);
+    turtle.forward(SIZE * 8.0);
+
+    turtle.end_fill();
+}
+
+fn draw_leg(turtle: &mut Turtle) {
+    turtle.begin_fill();
+
+    for _ in 0..15 {
+        turtle.forward(SIZE);
+        turtle.left(0.5);
+    }
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 6.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 3.0);
+
+    for _ in 0..90 {
+        turtle.forward(SIZE / 6.0);
+        turtle.left(1.0);
+    }
+
+    turtle.forward(SIZE * 14.5);
+
+    turtle.end_fill();
+}
+
+fn draw_neck(turtle: &mut Turtle) {
+    turtle.begin_fill();
+
+    for _ in 0..15 {
+        turtle.forward(SIZE * 3.0);
+        turtle.left(1.5);
+    }
+
+    turtle.end_fill();
+    turtle.begin_fill();
+
+    turtle.left(100.0);
+    turtle.forward(SIZE * 20.0);
+    turtle.left(80.0);
+
+    for _ in 0..4 {
+        turtle.forward(SIZE * 3.7);
+        turtle.left(1.5);
+    }
+
+    turtle.left(30.0);
+
+    for _ in 0..27 {
+        turtle.forward(SIZE);
+        turtle.right(1.0);
+    }
+
+    turtle.end_fill();
+}
+
+// In our case the most difficult part of the turtle is head. It requires six loops to draw.
+// There was a way to draw it as one circle but
+// I've decided to make it look a like a real turtle's head.
```

Comment:
This is great! These kinds of explanations of the rationale behind your decisions are extremely valuable! :smile: